tests/unit/commands/serve-test.js

```diff
@@ -54,7 +68,159 @@ describe('serve command', function() {
+        variations.push({inputOpts: { port: '0', host: '127.0.0.1' }, expectedOpts: { port: 0, family: 'IPv4', address: '127.0.0.1' }});
+        portZeroTries--;
+      }
+
+      return Promise.all(variations.map(function(variation) {
+        return testVariation(variation);
+      })).then(function() {
+        expect(tasks.Serve.prototype.run.called).to.equal(variations.length, 'expected run to be called '+variations.length);
+      });
+    });
+  }
+
+  // will eventually fail if you set the number of iterations high enough
+  var __testRuns = process.env.TEST_RUNS && Number(process.env.TEST_RUNS) || 1;
+  while(__testRuns > 0) {
+    __testRuns--;
+    crazyTest();
+  }
+
+  it('should fail if starting on a port that is bound to a different interface', function() {
+
+    return testServer({ port: '3000', host: '127.0.0.1' }, function(/*opts, server*/) {
+      return command.validateAndRun([
+        // commented out options throw when host ^^ is ::1
+        // AssertionError: has live reload port > port: expected 3000 to be within 49152..65535
+        // but why? and also the port should NOT ever be 3000 in these tests, that should fail
+        // anyway, here are way to break it
+
+
+        // '--port', '3000',
+        // '--live-reload-port', '3000'
+
+        // and these throw when testServer host is ::1 :)
+
+        // '--port', '3000',
+        // '--live-reload-port', '3000',
+        // '--host', '::1',
+        // '--live-reload-host', '::1'
+
+        // but this passes when testServer host is set to 0.0.0.0 or 127.0.0.1
+        // aka, ipv6 seems to mess up my testServer :( or something else, its late
+
+        '--port', '3000',
+        '--live-reload-port', '3000',
+        '--host', '0.0.0.0',
+        '--live-reload-host', '::1'
+
+        // In other words live reload test is working but line 147 of express-server
+        // is not being tested, which now that i write this, makes sense. time for sleep
+        // then. We don't run that express server, that'would be an acceptance test :P
+
+      ]).then(function() {
+        var serveRun = tasks.Serve.prototype.run;
+        var runOps = serveRun.calledWith[0][0];
+
+        expect(serveRun.called).to.equal(1, 'expected run to be called once');
+        expect(runOps.port).to.equal(3000, 'has correct port');
+        expect(runOps.liveReloadPort).to.be.within(49152, 65535, 'has live reload port > port');
+
+        // TODO: this test should throw. It does in real life. We've never tested
+        // the following error though yet:
+        //
+        // Example: say nginx is running on port 8000, this is what we
+        // want to test for, our expectation:
+        //
+        // ember serve -lrp 8000 -p 8000 -lrh ::1 -h ::1
+        // version: 2.3.0-beta.1-dev-984fefccda
+        // Could not serve on http://::1:8000. It is either in use or you do not have permission.
+      });
```

Comment:
i will cleanup these tests - truth is they're not really very accurate anyway, thoughts welcome

to actually test npm link to your project, start a server on a known port(or a couple one for express, another for live reload) then play with ember server -p PORT -lrp LIVERELOADPORT -h HOST -lrh LIVERELOADHOST

manually trying all the variations, including port 0 works. these tests though, they do lie and dont catch all edge cases, fyi